### PR TITLE
[minigraph] Add support for ipv4 dhcp_server in minigraph parser

### PR DESCRIPTION
**Pull request by yaqiangz**
#### Why I did it
Add minigraph parser support for dhcp_server in BmcMgmtToRRouter device followed https://github.com/sonic-net/SONiC/pull/1282. Parse dhcp_server related info from minigraph.

##### Work item tracking
- Microsoft ADO **(number only)**: 17695979

#### How I did it
Add minigraph parser support for dhcp_server in BmcMgmtToRRouter device.

#### How to verify it
Build image, ut passed.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by Blueve — APPROVED** [review]

**Comment by yaqiangz** [comment]
@yxieca Could you pls help to merge this PR?

**Comment by lguohan** [comment]
@qiluo-msft , i thought we are gong to block all minigraph changes now?

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -2068,6 +2140,18 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
...
+            generate_ipv4_dhcp_server_config(devices, link_ports, port_alias_map, vlan_intfs, vlan_members,
+                                             rack_mgmt_map)
+        for vlan_name, _ in results["VLAN"].items():
+            # For BmcMgmtToRRouter, treat docker ip as new dhcp_server
+            results["VLAN"][vlan_name]["dhcp_servers"] = [DHCP_SERVER_IP]
+        if dhcp_server_ipv4 is not None:
+            results['DHCP_SERVER_IPV4'] = dhcp_server_ipv4
+            if dhcp_server_ipv4_customized_options:
+                results['DHCP_SERVER_IPV4_CUSTOMIZED_OPTIONS'] = dhcp_server_ipv4_customized_options
+            results['DHCP_SERVER_IPV4_PORT'] = dhcp_server_ipv4_port
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:2154]: This is a new table in ConfigDB. Please onboard it with golden config generation pipeline. We are blocking new tables coming into minigraph parser.

**Review by qiluo-msft — CHANGES_REQUESTED** [review]
as comment

File: src/sonic-config-engine/minigraph.py [hunk 2]
```diff
@@ -2068,6 +2140,18 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
             'client_crt_cname': 'client.restapi.sonic'
         }
     }
+    if devices[hostname]["type"] == "BmcMgmtToRRouter":
+        dhcp_server_ipv4, dhcp_server_ipv4_port, dhcp_server_ipv4_customized_options = \
+            generate_ipv4_dhcp_server_config(devices, link_ports, port_alias_map, vlan_intfs, vlan_members,
```
Comment (qiluo-msft) [inline src/sonic-config-engine/minigraph.py:2145]: >generate_ipv4_dhcp_server_config [](http://example.com/codeflow?start=12&length=32)

Why generate_ipv4_dhcp_server_config() is needed. It just manipulated existing ConfigDB, does not parse minigraph at all, no extra info added into ConfigDB. Seems like everything you needed is already in ConfigDB or as default values.

**Pull request closed without merging**